Src/main.c

```diff
@@ -108,6 +109,11 @@ int main(void)
   /* USER CODE BEGIN WHILE */
   while (1)
   {
+    uint16_t adcOutput = 0;
+    adcOutput = HAL_ADC_GetValue(&hadc); //This will give a value corrosponding to 0-4095 and will be between 0-3.3v. It is a 12 bit long int.
+    uint16_t pwmPercentage = adcOutput/4095 * 100;
+    __HAL_TIM_SET_COMPARE(&htim16, TIM_CHANNEL_1, pwmPercentage);
```

Comment:
this macro writes your value directly to the 16 bit  capture/compare register (17.6.12 in the reference manual). You'll need to understand what the register does with the value you give it to know how you'll need to encode your duty cycle value. I can tell you that a percentage from 0-100 is not the right encoding.